src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -32,4 +34,22 @@
  */
 public interface Extent extends EntityUniverse, WeatherUniverse, BlockVolume, BiomeArea {
 
+    /**
+     * Get a representation of the block at the given position.
+     *
+     * @param position The position
+     * @return The block
+     */
+    BlockLoc getFullBlock(Vector3i position);
```

Comment:
Perhaps `BlockState getBlockSnapshot` and `BlockLoc getBlockExtended`? Describes it slightly better, I just dislike the wording of "full block."